Parse macro invocations containing $

ninevra opened the pull request:
Closes #95.

Adds a set of productions for general delimited token trees. These productions do not attempt to find `macro_rules!`-specific features like repetitions, and consider `$` to be a normal token. For backwards compatibility, use of the new productions aliases them to `token_tree`.

Blocks draft #93, which adds similar productions; #93 could be rebased on top of this if relevant.

aryx commented on grammar.js:
```diff
@@ -879,7 +879,23 @@ module.exports = grammar({
+
+    _non_delim_token: $ => choice(
+      $._non_special_token,
+      '$'
```
do we need other characters here? This is needed just for '$'?

ninevra replied:
I believe `$` is the only character we need to add here. `_non_special_token` is analogous to the first branch of the [`MacroMatch` construction](https://doc.rust-lang.org/reference/macros-by-example.html#macros-by-example) in the reference, which allows any token other than `$` and delimiters.

aryx reviewed (changes requested):
Could you rebase on the latest so there is no conflict?

ninevra commented:
Sorry, I've been away from github for a few months. I'll take a look at this and see if I can rebase & address comments soon.

ninevra commented:
Rebased to resolve merge conflicts.

Removed special handling of metavariables (`$ident`, `$crate`) in macro invocations, as those are specific to macro transcribers.

aryx commented:
This looks good! I guess this can unblock your other PRs now!

ninevra commented:
Thanks for merging! I'll work on rebasing #93.
